dataset/routers/v1/api_version/publish_version.py

```diff
@@ -137,11 +137,8 @@ async def download_dataset_files(self):
         for file in self.dataset_files:
             location_data = parse_minio_location(file['storage']['location_uri'])
             try:
-                await run_in_threadpool(
-                    self.mc.client.fget_object,
-                    location_data['bucket'],
-                    location_data['path'],
-                    self.tmp_folder + '/' + location_data['path'],
+                await self.s3_client.boto_client.download_object(
+                    location_data['bucket'], location_data['path'], self.tmp_folder + '/' + location_data['path']
```

Comment:
I would suggest to introduce e.g. a `download_file` method into `S3Client` and use it instead of accessing the internal `boto_client`.
```suggestion
                await self.s3_client.download_file(
                    location_data['bucket'], location_data['path'], self.tmp_folder + '/' + location_data['path']
```